designs/kubelet-cinfiguration.md

```diff
@@ -0,0 +1,203 @@
+        files:
+        - path: /etc/kubernetes/patches/kubeletconfiguration.yaml
+          owner: "root:root"
+          permissions: "0644"
+          content: |
+            apiVersion: kubelet.config.k8s.io/v1beta1
+            kind: KubeletConfiguration
+            address: "192.168.0.8"
+            port: 20250
+            serializeImagePulls: false
+            evictionHard:
+                memory.available:  "100Mi"
+                nodefs.available:  "10%"
+                nodefs.inodesFree: "5%"
+                imagefs.available: "15%"
+        initConfiguration:
+          patches:
+            directory: /etc/kubernetes/patches
+        joinConfiguration:
+          patches:
+            directory: /etc/kubernetes/patches
+
+Kubeadm config patches takes the default kubelet config that kubeadm would generate and overrides it with whatever fields are set in the patch files. This allows us to let users pass any subset of kubelet parameters they want to configure while also keeping the kubeadm defaults for other fields. This approach is explained in more detail in the CAPI doc here https://cluster-api.sigs.k8s.io/tasks/bootstrap/kubeadm-bootstrap/kubelet-config#use-kubeadms-kubeletconfiguration-patch-target
+
+
+#### Alternate Solution 
+
+Pass as string in the cluster spec
+
+```
+apiVersion: anywhere.eks.amazonaws.com/v1alpha1
+kind: Cluster
+metadata:
+  name: eksa-cluster
+spec:
+  controlPlaneConfiguration:
+    kubeletConfiguration: |
+      apiVersion: kubelet.config.k8s.io/v1beta1
+      kind: KubeletConfiguration
+      address: "192.168.0.8"
+      port: 20250
+      serializeImagePulls: false
+      evictionHard:
+          memory.available:  "100Mi"
+          nodefs.available:  "10%"
+          nodefs.inodesFree: "5%"
+          imagefs.available: "15%"
+```
+Pros
+
+* Simplicity of the input
+* Can be provided to the CAPI template as the direct string as that is the type of input CAPI would expect (no type conversions)
+
+Cons
+
+* EKSA will need additional yaml parsing validations to verify that its a valid yaml
+* Difficult to maintain for larger strings
+
+
+### Handling Kubelet params that EKS-A currently configures
+
+EKS-A spec currently already sets a few configs that are passed to kubelet through kubelet extra args. These include user configurable fields like node-labels and resolv-conf path, and also some EKS-A default ones like tls-cipher-suites, cloud-provider, read-only-port,  anonymous-auth and eviction-hard (some are specific to certain providers).
+There’s two questions that need we need to answer here:
+
+=> What happens when a user tries to configure a kubelet parameter directly and also through another cluster spec field?
+
+These are the current kubelet arguments we expose via our EKSA cluster spec today. 
+```
+   Resolv Conf (DNS) - clusterNetwork.dns.resolvConf
+   Node labels (CP and MD) - controlPlaneConfiguration.labels; workerNodeGroupConfigurations.labels
+   ```
+#### PROPOSED SOLUTION 
```

Comment:
I would recommend having a single section for proposed and alternate solutions. You can have multiple sub-sections within each of them for different scenarios/usecases